### PR TITLE
google-re2: init at 0.1.20210601

### PR DESCRIPTION
**Pull request by alexbakker**
###### Motivation for this change

New package. Depends on #99647.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alexbakker** [comment]
#99647 was merged.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99938` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python27Packages.google-re2</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.google-re2</li>
    <li>python38Packages.google-re2</li>
  </ul>
</details>

**Comment by alexbakker** [comment]
Apologies for the extremely long delay.

@SuperSandro2000 I resolved the merge conflict and bumped the version to the latest (0.1.20210601). I'm not sure if the Python 2 build failure is still a problem, as I don't think hydra still builds that set anymore anyway. But if it is a problem, how should I go about excluding Python 2 for this package?

File: pkgs/development/python-modules/google-re2/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage rec {
+  pname = "google-re2";
+  version = "0.1.20210601";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1f1ql95f97ss8i0rn1c37kgi0qrf1nq9b3q8xbq9x3gwg7xgzi71";
+  };
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/google-re2/default.nix:12]: This probably need to be propagatedBuildInputs.

**Comment by SuperSandro2000** [comment]
> Apologies for the extremely long delay.
> 
> @SuperSandro2000 I resolved the merge conflict and bumped the version to the latest (0.1.20210601). I'm not sure if the Python 2 build failure is still a problem, as I don't think hydra still builds that set anymore anyway. But if it is a problem, how should I go about excluding Python 2 for this package?

It is excluded now. You can disable certain versions with ``disabled = pythonAtLeast "3.6"``.

**Comment by alexbakker** [comment]
Addressed both comments.

File: pkgs/development/python-modules/google-re2/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, pythonAtLeast, fetchPypi, pybind11, re2, six }:
+
+buildPythonPackage rec {
+  pname = "google-re2";
+  version = "0.1.20210601";
+  disabled = !pythonAtLeast "3.6";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/google-re2/default.nix:6]:
```suggestion
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/google-re2/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, pythonAtLeast, fetchPypi, pybind11, re2, six }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/google-re2/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, pythonOlder, fetchPypi, pybind11, re2, six }:
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 99938` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.google-re2</li>
<li>python39Packages.google-re2</li>
</ul>
</details>

**Pull request merged**